File: baekjoon/java/src/silver5/Q14916_Change.java
package silver5;

import java.util.Scanner;

// https://www.acmicpc.net/problem/14916
public class Q14916_Change {
    public static void main(String[] args) {
        int n = new Scanner(System.in).nextInt();
        int maxFive = n / 5;
        int cnt = 0;
        while (maxFive >= 0) {
            int temp = n - maxFive * 5;
            if (temp % 2 == 0) {
                cnt += (maxFive + temp / 2);
                break;
            }
            maxFive--;
        }
        if (maxFive < 0) {
            System.out.println(-1);
        } else {
            System.out.println(cnt);
        }
    }
}

File: baekjoon/python/silver5/q14916_change.py
## https://www.acmicpc.net/problem/14916
def solution() :
    n = int(input())
    max_five = n // 5
    cnt = 0
    while max_five >= 0:
        temp = n 
        temp -= max_five * 5
        if(temp %2 == 0):
            cnt = max_five + temp // 2
            break
        max_five -= 1
    if max_five < 0 :
        return -1
    return cnt

print(solution())
